recipes/llvm-core/all/conanfile.py

```diff
@@ -1,429 +1,510 @@
-        self._patch_sources()
+        if cross_building(self):
+            # FIXME support cross compilation, at least for common cases like Apple Silicon -> X86
+            #  requires a host-compiled version of llvm-tablegen
+            raise ConanInvalidConfiguration("Cross compilation is not supported")
 
-    def build(self):
-        self._patch_build()
-        cmake = self._configure_cmake()
-        cmake.configure()
-        cmake.build()
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def _apply_resource_limits(self, cmake_definitions):
+        if os.getenv("CONAN_CENTER_BUILD_SERVICE"):
+            self.output.info("Applying CCI Resource Limits")
+            cmake_definitions.update({
+                "LLVM_RAM_PER_LINK_JOB": "16384",
+                "LLVM_RAM_PER_COMPILE_JOB": "2048"
+            })
+        else:
+            if self.options.ram_per_compile_job != "auto":
+                cmake_definitions["LLVM_RAM_PER_COMPILE_JOB"] = self.options.ram_per_compile_job
+            if self.options.ram_per_link_job != "auto":
+                cmake_definitions["LLVM_RAM_PER_LINK_JOB"] = self.options.ram_per_link_job
 
     @property
-    def _module_subfolder(self):
-        return os.path.join("lib", "cmake", "llvm")
+    def _targets_to_build(self):
+        return ";".join(target for target in LLVM_TARGETS if self.options.get_safe(f"with_target_{target.lower()}"))
 
     @property
-    def _alias_module_file_rel_path(self):
-        return os.path.join(self._module_subfolder, "conan-official-{}-targets.cmake".format(self.name))
+    def _all_targets(self):
+        # This is not just LLVM_TARGETS as it is version specific
+        return ";".join(
+            target for target in LLVM_TARGETS if self.options.get_safe(f"with_target_{target.lower()}") is not None)
 
     @property
-    def _old_alias_module_file_rel_path(self):
-        return os.path.join(self._module_subfolder, "conan-official-{}-old-targets.cmake".format(self.name))
-
-    def _create_cmake_module_alias_targets(self, module_file, targets):
-        content = ""
-        for alias, aliased in targets.items():
-            content += textwrap.dedent("""\
-                if(TARGET {aliased} AND NOT TARGET {alias})
-                    add_library({alias} INTERFACE IMPORTED)
-                    set_property(TARGET {alias} PROPERTY INTERFACE_LINK_LIBRARIES {aliased})
-                endif()
-            """.format(alias=alias, aliased=aliased))
-        save(self, module_file, content)
+    def _msvcrt(self):
+        msvcrt = str(self.settings.compiler.runtime)
+        # handle conan legacy setting
+        if msvcrt in ["MDd", "MTd", "MD", "MT"]:
+            return msvcrt
+
+        if self.settings.build_type in ["Debug", "RelWithDebInfo"]:
+            crt = {"static": "MTd", "dynamic": "MDd"}
+        else:
+            crt = {"static": "MT", "dynamic": "MD"}
+
+        return crt[msvcrt]
```

Comment:
Perhaps you can use `conan.tools.microsoft.visual.msvc_runtime_flag()` to simplify this logic?

Reply:
Good spot - I just didn't know it existed.